backend/eventloop/db_reconciler.go

```diff
@@ -0,0 +1,208 @@
+	}()
+
+}
+
+// databaseReconcile loops through the DTAMs in a database, and verifies they are still valid. If not, the resources are deleted.
+func databaseReconcile(ctx context.Context, dbQueries db.DatabaseQueries, client client.Client, log logr.Logger) {
+	offSet := 0
+
+	log = log.WithValues("job", "databaseReconciler")
+
+	// Continuously iterate and fetch batches until all entries of DeploymentToApplicationMapping table are processed.
+	for {
+		if offSet != 0 {
+			time.Sleep(sleepIntervalsOfBatches)
+		}
+
+		var listOfdeplToAppMapping []db.DeploymentToApplicationMapping
+
+		// Fetch DeploymentToApplicationMapping table entries in batch size as configured above.​
+		if err := dbQueries.GetDeploymentToApplicationMappingBatch(ctx, &listOfdeplToAppMapping, appRowBatchSize, offSet); err != nil {
+			log.Error(err, fmt.Sprintf("Error occurred in Database Reconcile while fetching batch from Offset: %d to %d: ",
+				offSet, offSet+appRowBatchSize))
+			break
+		}
+
+		// Break the loop if no entries are left in table to be processed.
+		if len(listOfdeplToAppMapping) == 0 {
+			log.Info("All DeploymentToApplicationMapping entries are processed by Namespace Reconciler.")
+			break
+		}
+
+		// Iterate over batch received above.
+		for i := range listOfdeplToAppMapping {
+			deplToAppMappingFromDB := listOfdeplToAppMapping[i] // To avoid "Implicit memory aliasing in for loop." error.
+			gitOpsDeployment := managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deplToAppMappingFromDB.DeploymentName,
+					Namespace: deplToAppMappingFromDB.DeploymentNamespace,
+				},
+			}
+
+			if err := client.Get(ctx, types.NamespacedName{Name: deplToAppMappingFromDB.DeploymentName, Namespace: deplToAppMappingFromDB.DeploymentNamespace}, &gitOpsDeployment); err != nil {
+				if apierr.IsNotFound(err) {
+					log.Info("GitOpsDeployment " + gitOpsDeployment.Name + " not found in Cluster, probably user deleted it, " +
+						"but It still exists in DB, hence deleteing related database entries.")
+
+					// If GitOpsDeployment CR is not found in cluster, delete related entries from table
+					err = cleanEntriesFromDb(ctx, &deplToAppMappingFromDB, dbQueries, log)
+					if err != nil {
+						log.Error(err, "Error occurred in Database Reconciler while cleaning gitOpsDeployment entries from DB: "+gitOpsDeployment.Name)
+					}
+				} else {
+					log.Error(err, "Error occurred in Database Reconciler while fetching GitOpsDeployment from cluster: "+gitOpsDeployment.Name)
+				}
+			}
+
+			log.Info("Database Reconcile processed deploymentToApplicationMapping entry: " + deplToAppMappingFromDB.Deploymenttoapplicationmapping_uid_id)
+		}
+
+		// Skip processed entries in next iteration
+		offSet += appRowBatchSize
+	}
+}
+
+// cleanEntriesFromDb deletes database entries related to a given GitOpsDeployment
+func cleanEntriesFromDb(ctx context.Context, deplToAppMapping *db.DeploymentToApplicationMapping,
```

Comment:
The existing `cleanOldGitOpsDeploymentEntry()` handles cleaning up old DB entries. It also takes care of creating an Operation to delete the Application CR. Could we reuse it here?

Reply:
Yes, I initially I wanted to reuse it, but it requires a receiver and to reuse it here I I have to add some unrelated code. So as discussed in stand up, I created a new function, to delete DB entries, rest will be taken care by cluster-agent controller and Application CR will be deleted.